math/check_prime.cpp

```diff
@@ -2,55 +2,85 @@
  * Copyright 2020 @author omkarlanghe
  *
  * @file
- * A simple program to check if the given number if prime or not.
- *
  * @brief
- * Reduced all possibilities of a number which cannot be prime.
- * Eg: No even number, except 2 can be a prime number, hence we will increment
- * our loop with i+6 jumping and check for i or i+2 to be a factor of the
- * number; if it's a factor then we will return false otherwise true after the
- * loop terminates at the terminating condition which is (i*i<=num)
+ * A simple program to check if the given number is a prime or not.
+ * https://en.wikipedia.org/wiki/Primality_test
+ * @details
+ * A prime number is any number that can be divided only by itself and 1. It must
+ * be positive and a whole number, therefore any prime number is part of the
+ * set of natural numbers. The majority of prime numbers are even numbers, with
+ * the exception of 2. This algorithm finds prime numbers using this information.
+ * additional ways to solve the prime check problem:
+ * https://cp-algorithms.com/algebra/primality_tests.html#practice-problems
```

Comment:
```suggestion
 * https://cp-algorithms.com/algebra/primality_tests.html#practice-problems
 * @author [Omkar Langhe](https://github.com/omkarlanghe)
```

Reply:
authorship is usually last in the documentation. i found their github link so we are adding that as it is necessary to give credit where credit is due